File: sdk/typescript/examples/runAndInvoice.ts
import { FluxRouteClient } from "../dist/esm/index.js";

async function main() {
  const client = new FluxRouteClient({ apiKey: "demo-key" });

  await client.createTenant("tenant-a");
  await client.run("demo/manifests/tenant-a.yaml");
  await client.addUsage("tenant-a", 42);

  const month = new Date().toISOString().slice(0, 7);
  const summary = await client.getBillingSummary(month);
  const invoice = await client.getInvoice("tenant-a");

  console.log("summary", summary);
  console.log("invoice", invoice);
}

void main();

File: sdk/python/examples/run_and_invoice.py
from __future__ import annotations

from datetime import datetime, timezone

from fluxroute_sdk import FluxRouteClient


client = FluxRouteClient(api_key="demo-key")

client.create_tenant("tenant-a")
client.run("demo/manifests/tenant-a.yaml")
client.add_usage("tenant-a", 42)

month = datetime.now(timezone.utc).strftime("%Y-%m")
summary = client.get_billing_summary(month)
invoice = client.get_invoice("tenant-a")

print("summary:", summary)
print("invoice:", invoice)
